src/components/Issues.js

```diff
@@ -0,0 +1,161 @@
+import anime from 'animejs';
+import { Icon } from '../ui/components';
+import { getCSSVariableAsObject, getCSSVariableAsNumber } from '../utils/CSSVariables';
+import { flattenIssues, getFieldId } from '../utils/issues';
+
+const scrollTo = (destination) => {
+  if (!destination) { return; }
+  const scroller = scrollparent(destination);
+  const scrollStart = scroller.scrollTop;
+  const destinationOffset = parseInt(destination.getBoundingClientRect().top, 10);
+  const scrollEnd = scrollStart + destinationOffset;
+
+  anime({
+    targets: scroller,
+    scrollTop: scrollEnd,
+    easing: getCSSVariableAsObject('--animation-easing-js'),
+    duration: getCSSVariableAsNumber('--animation-duration-fast-ms'),
+  });
+};
+
+class Issues extends Component {
+  static propTypes = {
+    show: PropTypes.bool,
+    issues: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+  };
+
+  static defaultProps = {
+    show: true,
+    issues: {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: true,
+    };
+
+    this.flatIssues = flattenIssues(props.issues);
+    this.issueRefs = {};
+  }
+
+  componentWillReceiveProps(newProps) {
+    const noIssues = isEmpty(newProps.issues);
+    const show = newProps.show;
+
+    this.flatIssues = flattenIssues(newProps.issues);
+
+    // when panel hidden by parent reset collapsed state
+    if (noIssues || !show) { this.setState({ open: true }); }
+  }
+
+  /**
+   * Because display information for fields is essentially stored in the dom
+   * we use that as our data source for the field labels in the issue list.
+   */
+  componentDidUpdate() {
+    // for each issue get friendly title from dom
+    this.flatIssues.forEach(({ field }) => {
+      const fieldId = getFieldId(field);
+      const targetField = document.querySelector(`#${fieldId}`);
+      const fieldName = targetField.getAttribute('data-name') || targetField.textContent;
+
+      if (!fieldName) { return; }
+
+      this.issueRefs[fieldId].textContent = fieldName;
+    });
+  }
+
+  setIssueRef = (el, fieldId) => {
+    this.issueRefs[fieldId] = el;
+  }
+
+  isVisible = () =>
+    this.props.show && this.flatIssues.length > 0;
+
+  isOpen = () =>
+    this.state.open;
+
+  handleClickTitleBar = () =>
+    this.setState({ open: !this.state.open });
+
+  handleClickIssue = (e) => {
+    e.preventDefault();
+
+    const link = e.target.getAttribute('href');
+    const destination = document.querySelector(link);
+    scrollTo(destination);
+  }
+
+  render() {
+    const issues = map(
+      this.flatIssues,
+      ({ field, issue }) => {
+        const fieldId = getFieldId(field);
+
+        return (
+          <li
+            key={fieldId}
+            className="issues__issue"
+          >
+            <a
+              href={`#${fieldId}`}
+              onClick={this.handleClickIssue}
+            >
+              <span ref={el => this.setIssueRef(el, fieldId)}>
```

Comment:
This span changes the event target ([L91](https://github.com/codaco/Architect/pull/213/files#diff-ec976aaee2548bdfb5fb02cc65e88f5dR91)); scrolling fails when I click on the text itself.